server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/variables/GroupItemNode.java

```diff
@@ -33,20 +37,50 @@
  */
 @Getter
 @ToString(callSuper = true)
-public class GroupItemNode extends VariableWithLevelNode {
+@EqualsAndHashCode(callSuper = true)
+public class GroupItemNode extends VariableWithLevelNode implements IUsageClause {
   private boolean redefines;
+  private UsageFormat usageFormat;
 
-  public GroupItemNode(Locality location, int level, String name, boolean global, boolean redefines) {
+  public GroupItemNode(
+      Locality location,
+      int level,
+      String name,
+      boolean global,
+      boolean redefines,
+      UsageFormat usageFormat) {
     super(location, level, name, VariableType.GROUP_ITEM, global);
     this.redefines = redefines;
+    this.usageFormat = usageFormat;
   }
 
   @Override
   public List<SyntaxError> process() {
     List<SyntaxError> errors = new ArrayList<>();
-    if (getChildren().stream().noneMatch(node -> node.getNodeType() == NodeType.VARIABLE))
+    if (isValidChildrenPresent()
+        && getChildren().stream().noneMatch(node -> node.getNodeType() == NodeType.VARIABLE))
       errors.add(getError(MessageTemplate.of(EMPTY_STRUCTURE_MSG, getName())));
     errors.addAll(super.process());
     return errors;
   }
+
+  // Group item can have usage clause.
+  // Case 1.
+  //        Working-Storage Section.
+  //       01 FOO POINTER GLOBAL.   <-- This is valid. Here FOO node has only 1 children of
+  //       01 FOO2 pointer GLOBAL.       node type VARIABLE_DEFINITION_NAME. Which should not
+  //       Procedure Division.           be considered for empty structure.
+  //
+  // Case 2.
+  //         WORKING-STORAGE SECTION.
+  //        01 PARENT.    <--- This is invalid
+  //        PROCEDURE DIVISION.
```

Comment:
It is a part of important class logic, I would recommend moving it to the class javadoc

Reply:
Done